lib/reek/context/ghost_context.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'code_context'
+require_relative 'singleton_method_context'
+
+module Reek
+  module Context
+    # Semi-transparent context to represent a metaclass while building the
+    # context tree. This context will not be part of the resulting tree, but
+    # will track context and visibility separately while building is in
+    # progress.
+    class GhostContext
```

Comment:
Even though it is not a real `Context`, shouldn't it still inherit from `CodeContext`? Right now this class is just dangling in a sea of contexts that all inherit from `CodeContext`....
Additionally, you require `CodeContext` in the first line of the file but do not use it anywhere, so right now I'm not sure what the intention is here :(